audit/conn/domainsocket.go

```diff
@@ -0,0 +1,77 @@
+//go:build linux || darwin
+
+package conn
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/microsoft/go-otel-audit/audit/conn/internal/writer"
+)
+
+// Compile check on interface implementation.
+var _ Audit = DomainSocketConn{}
+
+// DomainSocketConn represents a connection to a remote audit server via a unix domain socket
+// set to SOCK_STREAM. This implements conn.Audit interface.
+type DomainSocketConn struct {
+	*writer.Conn
+}
+
+// Type returns the type of the audit connection.
+func (DomainSocketConn) Type() Type {
+	return TypeDomainSocket
+}
+
+func (DomainSocketConn) private() {}
+
+type dsOptions struct {
+	path string
+}
+
+func (o *dsOptions) setDefaults() {
+	const (
+		asaSocket    = "/var/run/mdsd/asa/default_fluent.socket"
+		fluentSocket = "/var/run/mdsd/default_fluent.socket"
+	)
+	if o.path != "" {
+		return
+	}
+
+	o.path = asaSocket
+	_, err := os.Stat(o.path)
+	if err != nil {
+		o.path = fluentSocket
+	}
+}
+
+// DSOption is an option for NewDomainSocket.
+type DSOption func(*dsOptions) error
+
+// DSPath sets the path to the unix domain socket. This is usually automatically detected and should
+// only be overridden in very specific circumstances.
+func DSPath(path string) DSOption {
+	return func(o *dsOptions) error {
+		o.path = path
+		return nil
+	}
+}
+
+// NewDomainSocket creates a new connection to the remote audit server.
+func NewDomainSocket(options ...DSOption) (DomainSocketConn, error) {
```

Comment:
minor: DomainSocket's are local-only right? Is the correct terminology in this case really still `remote audit server`?

Disclaimer: I know very little about domain sockets so this could be 100% correct, asking for my knowledge than anything.

Reply:
Domain sockets are local only, a special type of file.  But, in essence, you are creating a connection to the remote audit server, through the agent.